compiler/rustc_mir_build/src/build/block.rs

```diff
@@ -117,25 +118,94 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                     if let Some(init) = initializer {
                         let init = &this.thir[*init];
                         let initializer_span = init.span;
-
                         unpack!(
                             block = this.in_opt_scope(
                                 opt_destruction_scope.map(|de| (de, source_info)),
                                 |this| {
                                     let scope = (*init_scope, source_info);
                                     this.in_scope(scope, *lint_level, |this| {
```

Comment:
move the body of this closure to a separate function in `matches/mod.rs`

Reply:
It is now moved to `matches/mod.rs`